crates/core_arch/src/x86/avx512f.rs

```diff
@@ -1633,6 +1634,97 @@ pub unsafe fn _mm512_mask_cmp_epi64_mask(
     transmute(r)
 }
 
+/// Returns vector of type `__m512d` with undefined elements.
+///
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm512_undefined_pd)
+#[inline]
+#[target_feature(enable = "avx512f")]
+// This intrinsic has no corresponding instruction.
+pub unsafe fn _mm512_undefined_pd() -> __m512d {
+    // FIXME: this function should return MaybeUninit<__m512d>
+    mem::MaybeUninit::<__m512d>::uninit().assume_init()
+}
+
+/// Returns vector of type `__m512` with undefined elements.
+///
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm512_undefined_ps)
+#[inline]
+#[target_feature(enable = "avx512f")]
+// This intrinsic has no corresponding instruction.
+pub unsafe fn _mm512_undefined_ps() -> __m512 {
+    // FIXME: this function should return MaybeUninit<__m512>
+    mem::MaybeUninit::<__m512>::uninit().assume_init()
```

Comment:
This is UB.

Reply:
The `undefined` intrinsics should return zero-initialized vectors like clang does.

Reply:
_mm_undefined_ps and _mm_256_undefined_p{d,s} do same thing, should i change them?

Reply:
Yes those should be fixed as well.